docs/adr/0009-uniffi-tagging.md

```diff
@@ -0,0 +1,166 @@
+# Handling `UniffiTag` and using types from other crates
+
+* Status: proposed
+* Deciders: Uniffi developers
+* Date: 2024-01-05
+
+## Context and Problem Statement
+
+UniFFI's various FFI-related traits `FfiConverter`, `Lift`, `Lower`, `LiftReturn`, `LowerReturn`, etc. all have a generic "tag" parameter.
+The code in `uniffi_macros::setup_scaffolding` generates a zero-sized `crate::UniffiTag` type for use in that parameter in other generated code.
+For example, when lowering a type we use `<#type as Lower<crate::UniffiTag>>::lower`.
+
+The reason for this is the so-called "Orphan rule", which prevents generated code from implemented a remote trait on a remote type.
+If the FFI traits did not have a generic parameter, then it would be impossible to generate an code that implements those traits for a type defined in another crate.
```

Comment:
```suggestion
If the FFI traits did not have a generic parameter, then it would be impossible to generate code that implements those traits for a type defined in another crate.
```